OpenBSD$

enable GameController API by posing as XBox360 controller, like Linux does

remapping for XBox360 Controller
FIX: unclear how to invert axis

Index: src/joystick/SDL_gamecontroller.c
--- src/joystick/SDL_gamecontroller.c.orig
+++ src/joystick/SDL_gamecontroller.c
@@ -683,20 +683,19 @@ ControllerMapping_t *SDL_PrivateGetControllerMapping(i
         mapping = s_pEmscriptenMapping;
     }
 #endif
-#ifdef __LINUX__
+#if defined(__LINUX__) || defined(__OpenBSD__)
     if (!mapping) {
         const char *name = SDL_JoystickNameForIndex(device_index);
         if (name) {
-            if (SDL_strstr(name, "Xbox 360 Wireless Receiver")) {
-                /* The Linux driver xpad.c maps the wireless dpad to buttons */
-                SDL_bool existing;
-                mapping = SDL_PrivateAddMappingForGUID(jGUID,
-"none,X360 Wireless Controller,a:b0,b:b1,back:b6,dpdown:b14,dpleft:b11,dpright:b12,dpup:b13,guide:b8,leftshoulder:b4,leftstick:b9,lefttrigger:a2,leftx:a0,lefty:a1,rightshoulder:b5,rightstick:b10,righttrigger:a5,rightx:a3,righty:a4,start:b7,x:b2,y:b3,",
+	/* The Linux driver xpad.c maps the wireless dpad to buttons */
+	SDL_bool existing;
+	mapping = SDL_PrivateAddMappingForGUID(jGUID,
+	// Mapped for XBox 360 controller as recognized by OpenBSD
+"none,Generic USB Joystick,a:b0,b:b1,x:b2,y:b3,back:b8,start:b9,leftstick:b10,rightstick:b11,leftshoulder:b4,rightshoulder:b5,dpup:h0.1,dpdown:h0.4,dpleft:h0.8,dpright:h0.2,leftx:a0,lefty:a1,rightx:a2,righty:a3,lefttrigger:b6,righttrigger:b7,",
                               &existing);
-            }
-        }
+	}
     }
-#endif /* __LINUX__ */
+#endif /* __LINUX__ || __OpenBSD__ */
 
     if (!mapping) {
         const char *name = SDL_JoystickNameForIndex(device_index);
